skimage/transform/integral.py

```diff
@@ -15,6 +15,9 @@ def integral_image(image, *, dtype=None):
     ----------
     image : ndarray
         Input image.
+    dtype : data-type, optional
+        Data type (Numpy dtype specifier) to be use for calculation, and for
+        output array `S`.  If None, default to at least double float precision.
```

Comment:
What does "at least" mean here?

Reply:
I guess it's technically possible that the image could be an higher-precision type, or `complex128`.   How about `default the more precise of (double float and the `image` dtype)`?

Reply:
I tweaked this in a bit in https://github.com/scikit-image/scikit-image/pull/7753/commits/cf9e3cda52806005aaca3e9d61bb50aa12ed2364 according to the suggestions here.